src/main/groovy/org/checkerframework/gradle/plugin/CheckerFrameworkPlugin.groovy

```diff
@@ -144,9 +147,35 @@ final class CheckerFrameworkPlugin implements Plugin<Project> {
 
     // Apply checker to project
     project.gradle.projectsEvaluated {
+
+      // Decide whether to use ErrorProne Javac once configurations have been populated.
+      def actualCFDependencySet = project.configurations.checkerFramework.getAllDependencies()
+              .matching({dep ->
+        dep.getName().equals("checker") && dep.getGroup().equals("org.checkerframework")})
+
+      def CFVersionString
+      if (actualCFDependencySet.size() == 0) {
+        CFVersionString = new JarFile(project.configurations.checkerFramework.asPath).getManifest().getMainAttributes().getValue('Implementation-Version')
```

Comment:
@kelloggm All the included tests pass when using the Manifest. Does this work in this location now?

@zhangjiangqige Does this solve the issue in plume-util?

Reply:
@wmdietl Yes this works with plume-util.
@kelloggm Does the long filename of zip error happen with this? When not setting the path to a local checkerframework and not setting a version in build.gradle, `project.configurations.checkerFramework.asPath` gives something like `~/.gradle/caches/modules-2/files-2.1/org.checkerframework/checker/2.8.2/3d4faf688e0aee6dda024ba2ff2c856a1ed7d9ad/checker-2.8.2.jar` which looks pretty long again...

Reply:
This does still cause the "zip file too long" error.

I suggest reverting back to the version of the code that I suggested, and then adding another check to determine whether the `checker` version being used is a local file. If it is, add a branch that uses this method. I'm not exactly sure how to do that, though.

Reply:
Can you give instructions for how to reproduce the "zip file too long" error?
Can you add it to the test suite?

Reply:
I've been testing it using a sample gradle project on my local machine. I've uploaded that sample to github here: https://github.com/kelloggm/sample-gradle-project

You'll need to change the version of the plugin on your machine to `3.12-SNAPSHOT` in the `build.gradle` file of the plugin, and then run `./gradlew publishToMavelLocal`. Then, you can see the error by running `./gradlew clean build -PuseCheckerFramework=true` in the sample project directory.

I'll see if I can figure out how to add it as a test case.

Reply:
Following up:

I realized that that project uses a custom 3p checker I'm working on. I tried to use the nullness checker instead (which is in the `nullness-checker` branch of the repo I linked) and can't reproduce the error.

So it seems likely the problem is caused by my checker in combination with this feature rather than the feature itself. It is possible that the local name of the JAR for the custom checker is too long, causing the failure.

I would still prefer a version that works with my custom checker, but if this is blocking you all we should go ahead and merge your version, and I will deal with my own problems.

Reply:
How about adding a `skipCFVersionCheck` or some such option? If given, it falls back to using `LIBRARY_VERSION`. Then you can use that flag when the zip file error occurs.
Merging this PR would unblock the Java 9+ transition.

Reply:
I've added such a check. I no longer object to merging this PR.

Reply:
I've tested and it works with plume-util, either not specifying `skipCFVersionCheck` or setting it to false.

Reply:
Great! Could you merge this and make a release? Then we can update plume-util to use this new version.
Thanks!

Reply:
I've made the 0.3.13 release.